### DISCUSSION TITLE
Customize (goa gen) output folder

### DISCUSSION THREAD

**Discussion by aenawi** [goadesign/goa#3302] [General]
Is it possible to customize the `goa gen` command to generate all files into a specific folder other than `gen`?

Currently I already have a `./gen` folder used for other frameworks as well, so if I regenerate Goa code, the folder will be deleted completely and recreated all over! Any other sub folders will be deleted as well! I want one common `./gen` directory, but Goa will add another sub `/gen/gen` folder, so the folder will look like this:

```
./gen/gen/<goa services subfolders>
```

Why does Goa force the `gen` subfolder name?!

What I'm expecting is that Goa will generate all services' folders under the defined directory using -o arg. For example:

```
$ go gen myproject/design -o ./gen/goa
```
Goa should create all services' directories under `./gen/goa` without adding another `gen` subfolder. Running the above command will help me keep my `dbmodels` under `gen` directory:
```
~/myproject/gen/goa
~/myproject/gen/dbmodels
```

Any hints please?

**Comment by raphael** [comment]
Hello,

Goa does not currently allow specifying a different name for the `gen` directory. Part of it is that the name of the directory defines the name of the Go package when import statements do not explicitly set it so making that configurable affects the generated code.
